Mathlib/Data/Complex/Basic.lean

```diff
@@ -372,10 +372,10 @@ variable {R : Type*} [SMul R ℝ]
 instance instSMulRealComplex : SMul R ℂ where
   smul r x := ⟨r • x.re - 0 * x.im, r • x.im + 0 * x.re⟩
 
-theorem smul_re (r : R) (z : ℂ) : (r • z).re = r • z.re := by simp [(· • ·), SMul.smul]
+theorem smul_re (r : R) (z : ℂ) : (r • z).re = r • z.re := by simp [HSMul.hSMul, SMul.smul]
```

Comment:
This is because there needs to be a special case in cdot notation in core to handle `leftact%`, like https://github.com/leanprover/lean4/blob/a7efe5b60e86b26fefd4795b46d66af369b97329/src/Lean/Elab/BuiltinNotation.lean#L255